### ISSUE TITLE
No way to set PATH variable for remix?

### ISSUE THREAD

**Issue by TheRook** [ethereum/remix-desktop#183]
Hey, it looks like Remix can't find anything on OSX because you forgot to allow someone to be able to update their PATH so that brew and npm can include their paths. 

If you care at all about people using your project you need to allow remix to be able to use binaries on OSX.

**Comment by bunsenstraat** [comment]
please clarify.

**Comment by TheRook** [comment]
If you have used a unix like system before you should know that the PATH
variable is how binaries are located by executables.

On OSX you arent looking for executables installed by brew, and in OSX brew
is howthe vast majority of people will install the requirements.

**Comment by bunsenstraat** [comment]
Please provide step by step instructions to reproduce the problem you are having. Thanks.

**Comment by TheRook** [comment]
Here are the reproduction steps;
Attempt to Compile and deploy any solidity script with remix desktop on
OSX.  It will always fail, on every OSX machine - and I know for a fact to
OSX machines can build and deploy using remix....

Because literally not a single person on stack overflow has fixed your bug.

Not one. Because you hardcoded your environment variables, and you are
asking me why it's not working.

**Comment by TheRook** [comment]
@bunsenstraat There is no way your app has ever built and deployed on osx. I'm telling you this has never happened - everyone i have talked to gives up and uses the browser.  

https://stackoverflow.com/questions/70858571/sh-hardhat-command-not-found-when-installing-hardhat

**Comment by bunsenstraat** [comment]
I am assuming from your reply @TheRook that you are talking about using a hardhat project on localhost using Remix Desktop with the 'Enable Hardhat Compilation' enabled? Is this the error you are seeing?
<img width="1547" alt="Screenshot 2023-02-24 at 08 30 26" src="https://user-images.githubusercontent.com/2787300/221120098-689a5ca0-328a-4682-bf29-121792b20462.png">

**Comment by TheRook** [comment]
> I am assuming from your reply @TheRook that you are talking about using a hardhat project on localhost using Remix Desktop with the 'Enable Hardhat Compilation' enabled? Is this the error you are seeing? <img alt="Screenshot 2023-02-24 at 08 30 26" width="1547" src="https://user-images.githubusercontent.com/2787300/221120098-689a5ca0-328a-4682-bf29-121792b20462.png"> Can you also forward me the stack overflow links that pertain to this issue users are having with Remix Desktop?

Correct, it also appears to affect Linux and other *nix systems that rely upon a PATH environment variable to resolve the location of hardhat.  You can pull the PATH environment variable from the caller, or more commonly IDEs like vscode and eclipse allow you to set this ENV variable so that binaries can be called.

This clearly has never worked even once, and has been ignored for years. Can you see why this may be causing some frustration in the community?

**Comment by bunsenstraat** [comment]
All right. It will be fixed soon.
